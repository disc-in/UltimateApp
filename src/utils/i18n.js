import { I18n } from 'i18n-js';
import * as Localization from 'expo-localization';

import en from './locales/en';
import fr from './locales/fr';

const i18n = new I18n({
  en,
  fr,
});

i18n.defaultLocale = 'en';
i18n.locale = Localization.locale;
i18n.fallbacks = true;
i18n.enableFallback = true;

export default i18n;
