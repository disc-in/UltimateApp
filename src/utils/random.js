import * as Crypto from 'expo-crypto';

export const generateRandomHex = (size = 10) =>
  [...Array(size)].map(() => Math.floor(Math.random() * 16).toString(16)).join('');

export const generateUuid = () => Crypto.randomUUID();
