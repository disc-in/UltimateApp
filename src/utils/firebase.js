import { Platform, InteractionManager } from 'react-native';
import Constants from 'expo-constants';
import { initializeApp, setLogLevel } from 'firebase/app';
import { getDatabase, ref, set, child, get } from 'firebase/database';

import { generateRandomHex } from './random';

// Work around issue `Setting a timer for long time`
// see: https://github.com/firebase/firebase-js-sdk/issues/97
const _setTimeout = global.setTimeout;
const _clearTimeout = global.clearTimeout;
const MAX_TIMER_DURATION_MS = 60 * 1000;
if (Platform.OS === 'android') {
  const timerFix = {};
  const runTask = (id, fn, ttl, args) => {
    const waitingTime = ttl - Date.now();
    if (waitingTime <= 1) {
      InteractionManager.runAfterInteractions(() => {
        if (!timerFix[id]) {
          return;
        }
        delete timerFix[id];
        fn(...args);
      });
      return;
    }

    const afterTime = Math.min(waitingTime, MAX_TIMER_DURATION_MS);
    timerFix[id] = _setTimeout(() => runTask(id, fn, ttl, args), afterTime);
  };

  global.setTimeout = (fn, time, ...args) => {
    if (MAX_TIMER_DURATION_MS < time) {
      const ttl = Date.now() + time;
      const id = '_lt_' + Object.keys(timerFix).length;
      runTask(id, fn, ttl, args);
      return id;
    }
    return _setTimeout(fn, time, ...args);
  };

  global.clearTimeout = (id) => {
    if (typeof id === 'string' && id.startsWith('_lt_')) {
      _clearTimeout(timerFix[id]);
      delete timerFix[id];
      return;
    }
    _clearTimeout(id);
  };
}
// End of workaround

const firebaseConfig = {
  databaseURL: Constants.expoConfig.extra.firebaseDatabaseUrl,
};

const app = initializeApp(firebaseConfig);
setLogLevel('silent');

const reference = (namespace, identifier) => {
  return ref(getDatabase(app), `${namespace}/${identifier}`);
};

export const upload = async (namespace, record) => {
  const withoutUndefineds = JSON.parse(JSON.stringify(record));
  const identifier = generateRandomHex();
  await set(reference(namespace, identifier), withoutUndefineds);
  return identifier;
};

export const download = (namespace, identifier) => {
  const dbRef = ref(getDatabase(app));
  return get(child(dbRef, `${namespace}/${identifier}`))
    .then((snapshot) => {
      if (snapshot.exists()) {
        return snapshot.val();
      } else {
        return null;
      }
    })
    .catch((error) => {
      console.error(error);
    });
};
