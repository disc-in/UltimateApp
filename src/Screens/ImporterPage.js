import React, { useState, useEffect } from 'react';
import { StyleSheet, View, Text } from 'react-native';
import { connect } from 'react-redux';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import { Input } from 'react-native-elements';

import theme from '../styles/theme.style';
import I18n from '../utils/i18n';
import { download } from '../utils/firebase';
import { showError } from '../utils/flashMessage';
import { savePlay } from '../Store/Actions/playAction';
import { saveDrill } from '../Store/Actions/drillAction';
import Button from '../Components/shared/Button';

export const ImporterPage = (props) => {
  const { navigation } = props;
  const namespace = props.route.params.source;

  const [identifier, setIdentifier] = useState();
  const [loading, setLoading] = useState(false);
  const [importedRecord, setImportedRecord] = useState();

  fetchData = async () => {
    try {
      const record = await download(namespace, identifier);
      if (record === null) {
        showError(I18n.t(`importerPage.${namespace}.downloadError`));
      } else {
        setImportedRecord(record);
      }
    } catch (error) {
      showError(I18n.t(`importerPage.${namespace}.downloadError`));
    }
  };

  const save = () => {
    if (namespace === 'customPlays') {
      props.savePlay(importedRecord);
      navigation.navigate('PlayEditorPage', { currentPlay: importedRecord });
    } else if (namespace === 'customDrills') {
      props.saveDrill(importedRecord);
      navigation.navigate('DrillPage', { id: importedRecord.id });
    }
  };

  const cancel = () => {
    setIdentifier();
    setLoading(false);
    setImportedRecord();
  };

  return (
    <View style={styles.importerPage}>
      {importedRecord ? (
        <View>
          <Text style={styles.text}>
            {I18n.t(`importerPage.${namespace}.incentive`, { title: importedRecord.title })}
          </Text>
          <Text style={styles.text}>{I18n.t(`importerPage.${namespace}.question`)}</Text>
          <View style={styles.ctaArea}>
            <Button style={styles.cta} text={I18n.t('shared.yes')} onPress={save} />
            <Button style={styles.cta} text={I18n.t('shared.cancel')} onPress={cancel} light />
          </View>
        </View>
      ) : loading ? (
        <>
          <MaterialCommunityIcons name="download" size={72} />
          <Text>{I18n.t(`importerPage.${namespace}.loading`)}</Text>
        </>
      ) : (
        <Input
          autoFocus
          value={identifier}
          onChangeText={setIdentifier}
          placeholder={I18n.t('importerPage.identifierPlaceholder')}
          testID="identifierInput"
          rightIcon={{
            type: 'material-community',
            name: 'check',
            testID: 'identifierInputSubmit',
            onPress: () => {
              setLoading(true);
              fetchData();
            },
          }}
        />
      )}
    </View>
  );
};

const mapDispatchToProps = { savePlay, saveDrill };

export default connect(null, mapDispatchToProps)(ImporterPage);

const styles = StyleSheet.create({
  importerPage: {
    backgroundColor: theme.BACKGROUND_COLOR_LIGHT,
    flex: 1,
    alignItems: 'center',
    padding: 20,
  },
  text: {
    fontSize: theme.FONT_SIZE_MEDIUM,
    marginBottom: 5,
  },
  ctaArea: {
    marginTop: 20,
    flexDirection: 'row',
    justifyContent: 'space-evenly',
  },
  cta: {
    width: 100,
  },
});
