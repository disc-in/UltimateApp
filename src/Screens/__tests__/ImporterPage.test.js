import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';

import * as firebase from '../../utils/firebase';
import * as flashMessage from '../../utils/flashMessage';
import { createDrill } from '../../Fixtures/TestFixtures';
import * as random from '../../utils/random';

import { ImporterPage } from '../ImporterPage';

describe('<ImporterPage />', () => {
  afterEach(() => jest.clearAllMocks());

  const play = {
    uuid: '123',
    animation: {
      positions: [[], []],
      ids: [],
      texts: [],
      background: 'endzone',
    },
    title: 'Title',
  };
  const drill = createDrill({ id: 1, custom: true });
  const navigation = { navigate: jest.fn() };
  const savePlay = jest.fn();
  const saveDrill = jest.fn();

  function itHandlesImport(route, identifier) {
    it('renders correctly before download', async () => {
      jest.spyOn(firebase, 'download').mockImplementation(() => new Promise((resolve) => resolve(undefined)));
      const { toJSON } = render(
        <ImporterPage navigation={navigation} route={route} savePlay={savePlay} saveDrill={saveDrill} />,
      );
      expect(toJSON()).toMatchSnapshot();
    });

    it('renders an error message if the play could not be found', async () => {
      jest.spyOn(firebase, 'download').mockImplementation(() => new Promise((resolve) => resolve(null)));
      jest.spyOn(flashMessage, 'showError');

      const { getByTestId, toJSON } = await waitFor(() =>
        render(<ImporterPage navigation={navigation} route={route} savePlay={savePlay} saveDrill={saveDrill} />),
      );

      fireEvent.changeText(getByTestId('identifierInput'), identifier);
      await fireEvent.press(getByTestId('identifierInputSubmit'));

      expect(flashMessage.showError).toHaveBeenCalled();
      expect(toJSON()).toMatchSnapshot();
    });

    it('renders correctly after download', async () => {
      jest.spyOn(firebase, 'download').mockImplementation(() => new Promise((resolve) => resolve(play)));

      const { getByTestId, toJSON } = await waitFor(() =>
        render(<ImporterPage navigation={navigation} route={route} savePlay={savePlay} saveDrill={saveDrill} />),
      );

      fireEvent.changeText(getByTestId('identifierInput'), identifier);
      await waitFor(() => fireEvent.press(getByTestId('identifierInputSubmit')));

      expect(toJSON()).toMatchSnapshot();
    });

    it('goes back to beginning on cancel', async () => {
      jest.spyOn(firebase, 'download').mockImplementation(() => new Promise((resolve) => resolve(play)));

      const { getByText, getByTestId, toJSON } = await waitFor(() =>
        render(<ImporterPage navigation={navigation} route={route} savePlay={savePlay} saveDrill={saveDrill} />),
      );

      fireEvent.changeText(getByTestId('identifierInput'), identifier);
      await waitFor(() => fireEvent.press(getByTestId('identifierInputSubmit')));

      await fireEvent.press(getByText('Cancel'));

      expect(toJSON()).toMatchSnapshot();
    });
  }

  describe('when routing on plays', () => {
    const route = {
      params: {
        source: 'customPlays',
      },
    };

    itHandlesImport(route, '123');

    it('saves play and opens editor on confirmation', async () => {
      jest.spyOn(firebase, 'download').mockImplementation(() => new Promise((resolve) => resolve(play)));
      jest.spyOn(random, 'generateUuid').mockImplementation(() => '123456');

      const { getByTestId, getByText } = await waitFor(() =>
        render(<ImporterPage navigation={navigation} route={route} savePlay={savePlay} saveDrill={saveDrill} />),
      );

      fireEvent.changeText(getByTestId('identifierInput'), '123');
      await waitFor(() => fireEvent.press(getByTestId('identifierInputSubmit')));
      await fireEvent.press(getByText('Yes'));

      const importedPlay = { ...play, uuid: '123456' };
      expect(savePlay).toHaveBeenCalledWith(importedPlay);
      expect(saveDrill).not.toHaveBeenCalled();
      expect(navigation.navigate).toHaveBeenCalledWith('PlayEditorPage', { currentPlay: importedPlay });
    });
  });

  describe('when routing on drills', () => {
    const route = {
      params: {
        source: 'customDrills',
      },
    };

    itHandlesImport(route, '123');

    it('saves play and opens editor on confirmation', async () => {
      jest.spyOn(firebase, 'download').mockImplementation(() => new Promise((resolve) => resolve(drill)));
      jest.spyOn(random, 'generateUuid').mockImplementation(() => '123456');

      const { getByTestId, getByText } = await waitFor(() =>
        render(<ImporterPage navigation={navigation} route={route} savePlay={savePlay} saveDrill={saveDrill} />),
      );
      fireEvent.changeText(getByTestId('identifierInput'), '123');
      await waitFor(() => fireEvent.press(getByTestId('identifierInputSubmit')));
      await fireEvent.press(getByText('Yes'));

      const importedDrill = { ...drill, id: '123456' };
      expect(savePlay).not.toHaveBeenCalled();
      expect(saveDrill).toHaveBeenCalledWith(importedDrill);
      expect(navigation.navigate).toHaveBeenCalledWith('DrillPage', { id: '123456' });
    });
  });
});
