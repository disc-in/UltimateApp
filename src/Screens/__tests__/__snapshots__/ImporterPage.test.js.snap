// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImporterPage /> when routing on drills goes back to beginning on cancel 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": [
            {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        autoFocus={true}
        editable={true}
        onChangeText={[Function]}
        placeholder="Use an identifier‚Ä¶"
        placeholderTextColor="#86939e"
        style={
          {
            "color": "#242424",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="identifierInput"
        underlineColorAndroid="transparent"
      />
      <View
        style={
          {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginVertical": 4,
            "paddingRight": 4,
          }
        }
      >
        <View
          style={
            {
              "overflow": "hidden",
            }
          }
        >
          <View
            accessibilityState={
              {
                "disabled": false,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            testID="identifierInputSubmit"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#242424",
                      "fontSize": 24,
                    },
                    {
                      "backgroundColor": "transparent",
                    },
                    {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
                testID="iconIcon"
              >
                Û∞Ñ¨
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <Text
      style={
        {
          "color": "#ff190c",
          "fontSize": 12,
          "margin": 5,
        }
      }
    />
  </View>
</View>
`;

exports[`<ImporterPage /> when routing on drills renders an error message if the play could not be found 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 72,
        },
        undefined,
        {
          "fontFamily": "material-community",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    Û∞áö
  </Text>
  <Text>
    I am fetching the drill you want to import it‚Ä¶ üèÉ
  </Text>
</View>
`;

exports[`<ImporterPage /> when routing on drills renders correctly after download 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View>
    <Text
      style={
        {
          "fontSize": 16,
          "marginBottom": 5,
        }
      }
    >
      You are about to import and save the drill Title
    </Text>
    <Text
      style={
        {
          "fontSize": 16,
          "marginBottom": 5,
        }
      }
    >
      Are you sure?
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "marginTop": 20,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#2a9e91",
            "borderRadius": 5,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 10,
            "width": 100,
          }
        }
        testID="button"
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Yes
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderColor": "#2a9e91",
            "borderRadius": 5,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 10,
            "width": 100,
          }
        }
        testID="button"
      >
        <Text
          style={
            {
              "color": "#2a9e91",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Cancel
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<ImporterPage /> when routing on drills renders correctly before download 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": [
            {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        autoFocus={true}
        editable={true}
        onChangeText={[Function]}
        placeholder="Use an identifier‚Ä¶"
        placeholderTextColor="#86939e"
        style={
          {
            "color": "#242424",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="identifierInput"
        underlineColorAndroid="transparent"
      />
      <View
        style={
          {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginVertical": 4,
            "paddingRight": 4,
          }
        }
      >
        <View
          style={
            {
              "overflow": "hidden",
            }
          }
        >
          <View
            accessibilityState={
              {
                "disabled": false,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            testID="identifierInputSubmit"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
    </View>
    <Text
      style={
        {
          "color": "#ff190c",
          "fontSize": 12,
          "margin": 5,
        }
      }
    />
  </View>
</View>
`;

exports[`<ImporterPage /> when routing on plays goes back to beginning on cancel 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": [
            {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        autoFocus={true}
        editable={true}
        onChangeText={[Function]}
        placeholder="Use an identifier‚Ä¶"
        placeholderTextColor="#86939e"
        style={
          {
            "color": "#242424",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="identifierInput"
        underlineColorAndroid="transparent"
      />
      <View
        style={
          {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginVertical": 4,
            "paddingRight": 4,
          }
        }
      >
        <View
          style={
            {
              "overflow": "hidden",
            }
          }
        >
          <View
            accessibilityState={
              {
                "disabled": false,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            testID="identifierInputSubmit"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#242424",
                      "fontSize": 24,
                    },
                    {
                      "backgroundColor": "transparent",
                    },
                    {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
                testID="iconIcon"
              >
                Û∞Ñ¨
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <Text
      style={
        {
          "color": "#ff190c",
          "fontSize": 12,
          "margin": 5,
        }
      }
    />
  </View>
</View>
`;

exports[`<ImporterPage /> when routing on plays renders an error message if the play could not be found 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 72,
        },
        undefined,
        {
          "fontFamily": "material-community",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    Û∞áö
  </Text>
  <Text>
    I am fetching the play you want to import it‚Ä¶ üèÉ
  </Text>
</View>
`;

exports[`<ImporterPage /> when routing on plays renders correctly after download 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View>
    <Text
      style={
        {
          "fontSize": 16,
          "marginBottom": 5,
        }
      }
    >
      You are about to import and save the play Title
    </Text>
    <Text
      style={
        {
          "fontSize": 16,
          "marginBottom": 5,
        }
      }
    >
      Are you sure?
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "marginTop": 20,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#2a9e91",
            "borderRadius": 5,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 10,
            "width": 100,
          }
        }
        testID="button"
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Yes
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderColor": "#2a9e91",
            "borderRadius": 5,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 10,
            "width": 100,
          }
        }
        testID="button"
      >
        <Text
          style={
            {
              "color": "#2a9e91",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Cancel
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<ImporterPage /> when routing on plays renders correctly before download 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": [
            {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        autoFocus={true}
        editable={true}
        onChangeText={[Function]}
        placeholder="Use an identifier‚Ä¶"
        placeholderTextColor="#86939e"
        style={
          {
            "color": "#242424",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="identifierInput"
        underlineColorAndroid="transparent"
      />
      <View
        style={
          {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginVertical": 4,
            "paddingRight": 4,
          }
        }
      >
        <View
          style={
            {
              "overflow": "hidden",
            }
          }
        >
          <View
            accessibilityState={
              {
                "disabled": false,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            testID="identifierInputSubmit"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
    </View>
    <Text
      style={
        {
          "color": "#ff190c",
          "fontSize": 12,
          "margin": 5,
        }
      }
    />
  </View>
</View>
`;
